python/pyarrow/tests/test_parquet.py

```diff
@@ -2668,3 +2668,40 @@ def test_parquet_file_too_small(tempdir):
         with open(path, 'wb') as f:
             f.write(b'ffff')
         pq.read_table(path)
+
+
+def test_multi_dataset_metadata(tempdir):
+    filenames = ["ARROW-1983-dataset.0", "ARROW-1983-dataset.1"]
+    metapath = str(tempdir / "_metadata")
+
+    # create a test dataset
+    df = pd.DataFrame({
+        'one': [1, 2, 3],
+        'two': [-1, -2, -3],
+        'three': [[1, 2], [2, 3], [3, 4]],
+        })
+    table = pa.Table.from_pandas(df)
+
+    # write dataset twice and collect/merge metadata
+    _meta = None
+    for filename in filenames:
+        meta = []
+        pq.write_table(table, str(tempdir / filename),
+                       metadata_collector=meta)
+        meta[0].set_file_path(filename)
+        if _meta is None:
+            _meta = meta[0]
+        else:
+            _meta.append_row_groups(meta[0])
+
+    # Write merged metadata-only file
```

Comment:
Should we test something about the actual `_meta` object here as well? (now it is only tested that the writing / reading roundtrip results in the same metadata). For example check the number of row groups? 
Although that is already covered in the C++ tests.

Reply:
I added some assert statements at the end to hopefully address this